common/docker/Dockerfile.training.unified

```
@@ -0,0 +1,94 @@
+ARG rapids_cuda_version="10.1"
+ARG rapids_release_type="runtime"
+ARG rapids_os="ubuntu18.04"
+ARG rapids_python="py3.8"
+FROM rapidsai/rapidsai:cuda${rapids_cuda_version}-${rapids_release_type}-${rapids_os}-${rapids_python}
+
+# ensure printed output/log-messages retain correct order
+ENV PYTHONUNBUFFERED=True
+
+RUN apt update -y \
+    && apt install -y --no-install-recommends build-essential \
+    && apt autoremove -y \
+    && apt clean -y
+
+# AWS Requirements
+ARG aws_dataset_directory="1_year"
+ARG aws_algorithm_choice="XGBoost"
+ARG aws_ml_workflow_choice="singleGPU"
+ARG aws_cv_folds="3"
+ARG aws_cloud_path="/opt/ml/code"
+#---------------------------------------------------
+ENV AWS_DATASET_DIRECTORY=${aws_dataset_directory}
+ENV AWS_ALGORITHM_CHOICE=${aws_algorithm_choice}
+ENV AWS_ML_WORKFLOW_CHOICE=${aws_ml_workflow_choice}
+ENV AWS_CV_FOLDS=${aws_cv_folds}
+ENV AWS_CLOUD_PATH=${aws_cloud_path}
+ENV RAPIDS_AWS_INSTALL_PATH=${aws_cloud_path}
+#===================================================
+
+## Azure Requirements
+ARG azure_install_path="/opt/rapids/azure"
+#----------------------------------------------
+ENV RAPIDS_AZURE_INSTALL_PATH=${azure_install_path}
+#==============================================
+
+## GCP Requirements
+ARG gcp_install_path="/opt/rapids/gcp"
+#----------------------------------------------
+ENV RAPIDS_GCP_INSTALL_PATH=${gcp_install_path}
+#==============================================
+
+# AWS Install
+RUN mkdir -p ${RAPIDS_AWS_INSTALL_PATH}
+ADD aws/code/entrypoint.sh \
+    aws/code/HPOConfig.py \
+    aws/code/HPODatasets.py \
+    aws/code/MLWorkflow.py \
+    aws/code/serve.py \
+    aws/code/train.py \
+    opt/rapids/aws/
+
+RUN . /opt/conda/etc/profile.d/conda.sh \
+    && conda activate rapids \
+    && conda install -c anaconda -c conda-forge \
+        flask \
+        dask-ml \
+    && pip install sagemaker-training
+
+# Azure Install
+RUN mkdir -p ${RAPIDS_AZURE_INSTALL_PATH}
+ADD azure/* \
+    /opt/rapids/azure/
+
+RUN . /opt/conda/etc/profile.d/conda.sh \
+    && conda activate rapids \
+    && pip install \
+        azureml-sdk \
+        azureml-widgets \
+        azureml-mlflow \
+        optuna \
+        dask-optuna
+
+# GCP Install
+RUN mkdir -p ${RAPIDS_GCP_INSTALL_PATH}
+ADD gcp/docker/infrastructure/* \
+    /opt/rapids/gcp/
+
+RUN . /opt/conda/etc/profile.d/conda.sh \
+    && conda activate rapids \
+    && pip install -U \
+        gcsfs \
+        cloudml-hypertune \
+        ax-platform \
+        sqlalchemy \
+        'ray[tune]'
+
+RUN mkdir -p /opt/rapids_cloudml
+ADD common/docker/infrastructure/* \
+     /opt/rapids_cloudml/
+
+## Unified entrypoint
+WORKDIR "/opt/rapids_cloudml"
+ENTRYPOINT [ "bash", "entrypoint.sh" ]
```

Adding /path/to/entrypoint fixes the issue:

```suggestion
ENTRYPOINT [ "bash", "/opt/rapids_cloudml/entrypoint.sh" ]
```
Looking at the docker run command from logs, -w overrides the Dockerfile WORKDIR